srcts/src/components/sidebar.ts

```diff
@@ -10,15 +10,50 @@ type MessageData = {
   method: SidebarMethod;
 };
 
-type SidebarComponents = {
+interface SidebarComponents {
   container: HTMLElement;
   main: HTMLElement;
   sidebar: HTMLElement;
   toggle: HTMLElement;
-  isClosed: boolean;
-};
+}
 
 class Sidebar {
+  private layout: SidebarComponents;
+  #isClosed = false;
+
+  constructor(container: HTMLElement) {
+    container.removeAttribute("data-bslib-sidebar-init");
+
+    Sidebar.instanceMap.set(container, this);
+    this.layout = {
+      container,
+      main: container.querySelector(":scope > .main") as HTMLElement,
+      sidebar: container.querySelector(":scope > .sidebar") as HTMLElement,
+      toggle: container.querySelector(
+        ":scope > .collapse-toggle"
+      ) as HTMLElement,
+    } as SidebarComponents;
+
+    if (!this.layout.toggle) {
+      throw new Error("Tried to initialize a non-collapsible sidebar.");
+    }
+
+    this.#isClosed = this._isClosedOrClosing();
+
+    this._initEventListeners();
+    this._initSidebarCounters();
+    this._initDesktop();
+  }
+
+  get isClosed(): boolean {
+    return this.#isClosed;
+  }
+
+  private _isClosedOrClosing(): boolean {
+    // While in the middle of toggling, the user may want to reverse course
```

Comment:
It would help to have a more detailed comment about the lifecycle of how classes are added removed (maybe just point to that code?)

Reply:
I added JSDoc strings to everything in Sidebar in 022fd4bdfbe222df5a4ba57540c4423cc7c62f2b (other than the input binding methods) and I think it's a really ties everything together nicely. In particular, I documented this lifecycle in the `.isClosed()` get method, since that's the public place where you'd be accessing state and wondering what's going on.

Reply:
Great, thanks!